### PR TITLE
Add Database Log table to query database logs for a workspace.

### PR DESCRIPTION
**Pull request by sidr0cker**
# Example query results
<details>
  <summary>Results</summary>
  
```
Add example SQL query results here (please include the input queries as well)
```
</details>

### CODE REVIEW

File: steampipecloud/table_steampipecloud_db_log.go [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+	openapi "github.com/turbot/steampipe-cloud-sdk-go"
+
+	"github.com/turbot/steampipe-plugin-sdk/v3/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/v3/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableSteampipeCloudWorkspaceDbLog(_ context.Context) *plugin.Table {
```
Comment (LalitLab) [inline steampipecloud/table_steampipecloud_db_log.go]:
```suggestion
func tableSteampipeCloudWorkspaceDBLog(_ context.Context) *plugin.Table {
```

File: steampipecloud/plugin.go [hunk 2]
```diff
@@ -30,6 +30,7 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 			"steampipecloud_workspace_connection":          tableSteampipeCloudWorkspaceConnection(ctx),
 			"steampipecloud_workspace_mod":                 tableSteampipeCloudWorkspaceMod(ctx),
 			"steampipecloud_workspace_mod_variable":        tableSteampipeCloudWorkspaceModVariable(ctx),
+			"steampipecloud_workspace_db_log":              tableSteampipeCloudWorkspaceDbLog(ctx),
```
Comment (LalitLab) [inline steampipecloud/plugin.go]:
```suggestion
			"steampipecloud_workspace_db_log":              tableSteampipeCloudWorkspaceDBLog(ctx),
```

File: steampipecloud/table_steampipecloud_db_log.go [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+	return &plugin.Table{
+		Name:        "steampipecloud_workspace_db_log",
+		Description: "Database logs records the underlying queries executed when a user executes a query.",
+		List: &plugin.ListConfig{
+			ParentHydrate: listWorkspaces,
+			Hydrate:       listWorkspaceDbLogs,
```
Comment (LalitLab) [inline steampipecloud/table_steampipecloud_db_log.go]:
```suggestion
			Hydrate:       listWorkspaceDBLogs,
```

File: steampipecloud/table_steampipecloud_db_log.go [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+				Description: "The time when the db log record was generated.",
+				Type:        proto.ColumnType_STRING,
+			},
+		},
+	}
+}
+
+//// LIST FUNCTION
+
+func listWorkspaceDbLogs(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```
Comment (LalitLab) [inline steampipecloud/table_steampipecloud_db_log.go]:
```suggestion
func listWorkspaceDBLogs(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
```

File: steampipecloud/table_steampipecloud_db_log.go [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+	}
+
+	if err != nil {
+		plugin.Logger(ctx).Error("listDbLogs", "error", err)
+		return nil, err
+	}
+	return nil, nil
+}
+
+func listUserWorkspaceDbLogs(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData, svc *openapi.APIClient, maxResults int32, identityId, workspaceId string) error {
```
Comment (LalitLab) [inline steampipecloud/table_steampipecloud_db_log.go]:
```suggestion
func listUserWorkspaceDBLogs(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData, svc *openapi.APIClient, maxResults int32, identityId, workspaceId string) error {
```

File: steampipecloud/table_steampipecloud_db_log.go [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+			pagesLeft = false
+		} else {
+			resp.NextToken = result.NextToken
+		}
+	}
+
+	return nil
+}
+
+func listOrgWorkspaceDbLogs(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData, svc *openapi.APIClient, maxResults int32, identityId, workspaceId string) error {
```
Comment (LalitLab) [inline steampipecloud/table_steampipecloud_db_log.go]:
```suggestion
func listOrgWorkspaceDBLogs(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData, svc *openapi.APIClient, maxResults int32, identityId, workspaceId string) error {
```

File: steampipecloud/table_steampipecloud_db_log.go [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+				maxResults = int32(1)
+			} else {
+				maxResults = int32(*limit)
+			}
+		}
+	}
+
+	// If we want to get the db logs for the user
+	if user.Id == workspace.IdentityId {
+		err = listUserWorkspaceDbLogs(ctx, d, h, svc, maxResults, user.Id, workspace.Id)
```
Comment (LalitLab) [inline steampipecloud/table_steampipecloud_db_log.go]:
```suggestion
		err = listUserWorkspaceDBLogs(ctx, d, h, svc, maxResults, user.Id, workspace.Id)
```

File: steampipecloud/table_steampipecloud_db_log.go [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+	} else {
+		err = listOrgWorkspaceDbLogs(ctx, d, h, svc, maxResults, workspace.IdentityId, workspace.Id)
```
Comment (LalitLab) [inline steampipecloud/table_steampipecloud_db_log.go]:
```suggestion
		err = listOrgWorkspaceDBLogs(ctx, d, h, svc, maxResults, workspace.IdentityId, workspace.Id)
```

**Review by LalitLab — CHANGES_REQUESTED** [review]
@siddharthaturbot if we could replace `Db` with `DB` in the functions name.
Otherwise, PR looks good

**Review by LalitLab — APPROVED** [review]
LGTM

File: steampipecloud/table_steampipecloud_organization.go [hunk 9]
```diff
@@ -110,7 +110,7 @@ func listOrganizations(ctx context.Context, d *plugin.QueryData, h *plugin.Hydra
 	// execute list call
 	pagesLeft := true
 
-	var resp openapi.ListActorOrgsResponse
+	var resp openapi.ListUserOrgsResponse
```
Comment (cbruno10) [inline steampipecloud/table_steampipecloud_organization.go:113]: @siddharthaturbot Is this change related to the table that this PR is adding? If not, can you please describe why we're making this change?

File: steampipecloud/table_steampipecloud_workspace_mod.go [hunk 10]
```diff
@@ -312,7 +308,6 @@ func getUserWorkspaceMod(ctx context.Context, d *plugin.QueryData, h *plugin.Hyd
 	response, err := plugin.RetryHydrate(ctx, d, h, getDetails, &plugin.RetryConfig{ShouldRetryError: shouldRetryError})
 
 	workspaceMod := response.(openapi.WorkspaceMod)
-	workspaceMod.Workspace = &openapi.Workspace{}
```
Comment (cbruno10) [inline steampipecloud/table_steampipecloud_workspace_mod.go:315]: Is this change related to the table we're adding?

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@siddharthaturbot Please see review questions, thanks!

File: steampipecloud/table_steampipecloud_organization.go (hunk 9 above)
Comment (sidr0cker) [reply steampipecloud/table_steampipecloud_organization.go:113]: ListActorOrgsResponse is not used by the SDK anymore. Its necessary to add it in the PR or it wont compile.

File: steampipecloud/table_steampipecloud_workspace_mod.go (hunk 10 above)
Comment (sidr0cker) [reply steampipecloud/table_steampipecloud_workspace_mod.go:315]: Same as above, Workspace is not part of the struct anymore.

**Review by cbruno10 — APPROVED** [review]
